org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/proxy/ConfigurableProxySelector.java

```diff
@@ -188,4 +192,24 @@ public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
     LOG.error("Connection to {} using {} as proxy failed.", uri, sa, ioe);
   }
 
+  /**
+   * Install an instance of {@link ConfigurableProxySelector} as default using
+   * {@link ProxySelector#setDefault(ProxySelector)}
+   */
+  public static class ProxySelectorInstaller implements IPlatformListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProxySelectorInstaller.class);
+
+    @Override
+    public void stateChanged(PlatformEvent event) {
```

Comment:
Inspired by [MultiSessionCookieStoreInstallListener ](https://github.com/eclipse-scout/scout.rt/blob/c67fb70429f99c3334992bfe596885d65ab9f51c/org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/servicetunnel/http/MultiSessionCookieStoreInstallListener.java#L18) which is just [always inactive](https://github.com/eclipse-scout/scout.rt/blob/c67fb70429f99c3334992bfe596885d65ab9f51c/org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/servicetunnel/http/MultiSessionCookieStoreInstallListener.java#L23) (non-configurable).

However I decided not to implement the uninstall (restore previous state) on bean manager stopped as with Jetty I suppose I application cannot be restarted anymore (the platform cannot be restarted)?

The [MultiSessionCookieStoreInstaller](https://github.com/eclipse-scout/scout.rt/blob/c67fb70429f99c3334992bfe596885d65ab9f51c/org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/servicetunnel/http/MultiSessionCookieStoreInstaller.java#L59) even includes code to ensure there is only one installation which is probably also not necessary if all our applications are run on our own using Jetty?

Back to topic (proxies): If the proxy selector is now installed as default it is even arguable if we should create other instances for our other HTTP clients (e.g. Apache/OkHttp) - or if we should just use system default here?